File: jiwook/BFSDFS/타겟 넘버.py
def solution(numbers, target):
    # +, - 부호에 따른 각 숫자에서 2가지 경우의 수
    def dfs(idx, current_sum):
            # 모든 숫자를 다 사용했을 때
            if idx == len(numbers):
                # 현재 합계가 타겟과 같다면 1, 아니면 0
                return 1 if current_sum == target else 0

            # 현재 숫자를 더하는 경우의 수 + 현재 숫자를 빼는 경우의 수
            plus = dfs(idx + 1, current_sum + numbers[idx])
            minus = dfs(idx + 1, current_sum - numbers[idx])

            return plus + minus
        # idx 0부터 시작, 초기 합계는 0
    return dfs(0, 0)

File: dayeong/PGS/깊이너비 우선 탐색(DFS, BFS)/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;

int dfs(vector<int> numbers, int target, int sum, int index){
    int answer = 0; //방법의 수
    
    if(index == numbers.size()){
        if(sum == target){
            return 1;
        }
        else{
            return 0;
        }
    }
    answer += dfs(numbers, target, sum + numbers[index], index+1);
    answer += dfs(numbers, target, sum - numbers[index], index+1);
    
    return answer;
}

int solution(vector<int> numbers, int target) {    
    return dfs(numbers, target, 0, 0);
}